sdk/appconfiguration/app-configuration/src/appConfigurationClient.ts

```diff
@@ -68,20 +72,43 @@ export class AppConfigurationClient {
    * Initializes a new instance of the AppConfigurationClient class.
    * @param connectionString Connection string needed for a client to connect to Azure.
    */
-  constructor(connectionString: string) {
-    const regexMatch = connectionString.match(ConnectionStringRegex);
-    if (regexMatch) {
-      const appConfigCredential = new AppConfigCredential(regexMatch[2], regexMatch[3]);
-      this.client = new AppConfiguration(appConfigCredential, apiVersion, {
-        baseUri: regexMatch[1],
+  constructor(connectionString: string);
+  /**
+   * Initializes a new instance of the AppConfigurationClient class using 
+   * a TokenCredential
```

Comment:
```suggestion
   * a TokenCredential.
```

Reply:
Muwahahahah

It's just horrible when sentences never end